src/sonic-yang-models/yang-models/sonic-pbh.yang

```diff
@@ -0,0 +1,218 @@
+					mandatory true;
+					type inet:ipv6-address;
+					when "boolean(../hash_field[.='INNER_DST_IPV6' or .='INNER_SRC_IPV6'])";
+				}
+
+				leaf sequence_id {
+					description "Configures in which order the fields are hashed and defines which fields should be associative";
+					mandatory true;
+					type uint32;
+				}
+
+			}
+			/* end of PBH_HASH_LIST */
+		}
+		/* end of container PBH_HASH */
+
+		container PBH_RULE {
+
+			description "PBH_RULE part of config_db.json";
+
+			list PBH_RULE_LIST {
+
+				key "table_name rule_name";
+
+				leaf table_name {
+					description "The name of table which holds this rule";
+					type leafref {
+						path "/pbh:sonic-pbh/pbh:PBH_TABLE/pbh:PBH_TABLE_LIST/pbh:table_name";
+					}
+				}
+
+				leaf rule_name {
+					description "The name of this rule";
+					type string;
+				}
+
+				leaf priority {
+					description "Configures priority for this rule";
+					mandatory true;
+					type uint32;
+				}
+
+				leaf gre_key {
+					description "Configures packet match for this rule: GRE key";
+					type string {
+						pattern "(0x){1}[a-fA-F0-9]+/(0x){1}[a-fA-F0-9]+";
+					}
+				}
+
+				leaf ip_protocol {
+					description "Configures packet match for this rule: IP protocol";
+					type string {
+						pattern "(0x){1}[a-fA-F0-9]+/(0x){1}[a-fA-F0-9]+";
+					}
+				}
+
+				leaf ipv6_next_header {
+					description "Configures packet match for this rule: IPv6 Next header";
+					type string {
+						pattern "(0x){1}[a-fA-F0-9]+/(0x){1}[a-fA-F0-9]+";
+					}
+				}
+
+				leaf l4_dst_port {
+					description "Configures packet match for this rule: L4 destination port";
+					type string {
+						pattern "(0x){1}[a-fA-F0-9]+/(0x){1}[a-fA-F0-9]+";
+					}
+				}
+
+				leaf inner_ether_type {
+					description "Configures packet match for this rule: inner EtherType";
+					type string {
+						pattern "(0x){1}[a-fA-F0-9]+/(0x){1}[a-fA-F0-9]+";
+					}
+				}
+
+				leaf-list hash_list {
+					description "The list of hash fields to apply with this rule";
+					min-elements 1;
+					type leafref {
+						path "/pbh:sonic-pbh/pbh:PBH_HASH/pbh:PBH_HASH_LIST/pbh:hash_name";
+					}
+				}
+
+				leaf packet_action {
+					description "Configures packet action for this rule";
+					type string {
+						pattern "SET_ECMP_HASH|SET_LAG_HASH";
+					}
+					default "SET_ECMP_HASH";
+				}
+
+				leaf flow_counter {
+					description "Enables/Disables packet/byte counter for this rule";
+					type string {
+						pattern "DISABLED|ENABLED";
+					}
+					default "DISABLED";
+				}
+
+			}
+			/* end of PBH_RULE_LIST */
+		}
+		/* end of container PBH_RULE */
+
+		container PBH_TABLE {
+
+			description "PBH_TABLE part of config_db.json";
+
+			list PBH_TABLE_LIST {
+
+				key "table_name";
+
+				leaf table_name {
+					description "The name of this table";
+					type string;
+				}
+
+				leaf-list port_list {
+					description "Ports to which this table is applied";
```

Comment:
Is it possible to have a single list for both PORT & PORTCHANNEL tables? also, why do we need default "" value? cant we  just avoid this field configuration itself?

Reply:
@venkatmahalingam Yes, it is possible. The current implementation is by design. The default "" value means that table can be created without ports/lags. The configuration can be added later via update.

Reply:
This will confuse the user, we could very well not have the field itself.

Reply:
@venkatmahalingam ok. We can merge.